### PR TITLE
Linear warmup learning rate schedule

### PR DESCRIPTION
**Pull request by srihari-humbarwadi**
# Description
Implements  linear warmup with step decay schedule, discussed in #2086

## Type of change
- [x] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running pre-commit hooks
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
I will be adding the tests upon feedback from the maintainers.

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Review by Harsh188 — CHANGES_REQUESTED** [review]
Can you add `tensorflow_addons/optimizers/piecewise_constant_decay_with_warmup.py` to [`source_code_test.py`](https://github.com/tensorflow/addons/blob/e2006c3e9a35c87649f0c902f90112b188871092/tools/testing/source_code_test.py#L115) on line 118.

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
